src/main/java/org/opentripplanner/util/time/ServiceDateUtils.java

```diff
@@ -0,0 +1,146 @@
+package org.opentripplanner.util.time;
+
+import java.text.ParseException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ServiceDateUtils {
+
+  private static final String MAX_TEXT = "MAX";
+  private static final String MIN_TEXT = "MIN";
+  private static final Pattern PATTERN = Pattern.compile("^(\\d{4})-?(\\d{2})-?(\\d{2})$");
+  private static final DateTimeFormatter COMPACT_FORMATTER = DateTimeFormatter.ofPattern(
+    "uuuuMMdd"
+  );
+
```

Comment:
Maybe leave a comment here that the strange-looking arithmetic is because of DST changes.